tests/forte/processors/stave_processor_test.py

```diff
@@ -0,0 +1,196 @@
+from forte.processors.base import PackProcessor, FixedSizeBatchProcessor
+from forte.processors.base.batch_processor import Predictor
+from ft.onto.base_ontology import Token, Sentence, EntityMention, RelationLink
+
+# Currently hard coded. Deprecated in future update.
+os.environ["FRONTEND_BUILD_PATH"] = os.path.abspath("stave/build/")
+sys.path.insert(0, os.path.abspath("stave/simple-backend/"))
+from forte.processors.stave import StaveProcessor
+from nlpviewer_backend.lib.stave_project import StaveProjectReader
+
+
+class TestStaveProcessor(unittest.TestCase):
+
+    def setUp(self):
+
+        self._port: int = 8880
+        self._file_dir_path = os.path.dirname(__file__)
+        self._project_name: str = "serialization_pipeline_test"
+        self._dataset_dir: str = os.path.abspath(os.path.join(
+            self._file_dir_path, '../../../', 'data_samples/ontonotes/00/'))
+        self._stave_processor = StaveProcessor()
+
+        self.pl = Pipeline[DataPack](
+            ontology_file=os.path.abspath(os.path.join(
+                self._file_dir_path, "../../../",
+                    "forte/ontology_specs/base_ontology.json"))
+        )
+        self.pl.set_reader(OntonotesReader())
+
+    def test_stave_viewer(self):
+        """
+        Test in viewer mode Stave
+        """
+        self.pl.add(self._stave_processor, config={
+            "project_name": self._project_name,
+            "server_thread_daemon": True
+        })
+        self.pl.run(self._dataset_dir)
+
+        project_reader = StaveProjectReader(
+            project_path=self._stave_processor.configs.project_path
+        )
+
+        self.assertEqual(project_reader.project_name, self._project_name)
+        self.assertEqual(
+            project_reader.project_type,
+            self._stave_processor.configs.project_type
+        )
+        self.assertEqual(
+            project_reader.ontology,
+            self._stave_processor.resources.get("onto_specs_dict")
+        )
+        self.assertEqual(
+            project_reader.project_configs,
+            self._stave_processor.configs.project_configs.todict()
+        )
+
+        # Check default project configuration
+        with open(os.path.abspath(os.path.join(
+            self._file_dir_path, "../data/ontology/test_specs/",
+            "test_project_configuration.json")), "r") as f:
+            target_configs = json.load(f)
+        self.assertEqual(
+            target_configs,
+            project_reader.project_configs,
+        )
+
+        # Check the number of newly created documents
+        count, index = 0, 0
+        while True:
+            next_index = project_reader.get_next_index(index)
+            if next_index == index:
+                break
+            count += 1
+            index = next_index
+
+        self.assertEqual(count + 1, len(os.listdir(self._dataset_dir)))
+
+    def test_stave_standard(self):
+        """
+        Test in standard Stave
```

Comment:
let's specify the purpose of the test a bit better, so future maintainer knows what to do.